src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -43,15 +41,9 @@ public bool TryGetProperty(string propertyName, out string value)
 
         public string GetProperty(string propertyName)
         {
-            if (Properties[propertyName].JTokenType != JTokenType.String)
+            if (!Properties[propertyName].IsString)
```

Comment:
As we discussed, we don't have to remember the "token type"; we just have to remember if we started out with a string.